src/internal/connector/onedrive/restore.go

```diff
@@ -45,6 +46,8 @@ func RestoreCollections(
 		metrics        support.CollectionMetrics
 		folderPerms    map[string][]UserPermission
 		canceled       bool
+		parentPerms    []UserPermission
+		ok             bool
```

Comment:
these variables only exist within the loop.  declaring them outside risks unwanted behavior.  safer to only declare them within the loop.